### PR TITLE
Add a section on increasing memory allocation to the MI300A system op…

### PR DESCRIPTION
**Pull request by amd-jnovotny**
…timization guide

### CODE REVIEW

File: docs/how-to/system-optimization/mi300a.rst [hunk 1]
```diff
@@ -122,6 +122,51 @@ This section describes performance-based settings.
 
        transparent_hugepage=always
 
+* **Increase the amount of allocatable memory**
+
+  By default, when using a device allocator via HIP, it is only possible to allocate 96GB out of 
+  a possible 128GB of memory on the MI300A. This limitation does not affect host allocations.
```
Comment (ckime-amd) [inline docs/how-to/system-optimization/mi300a.rst:128]: Should 96 GiB and 128 GiB be used for consistency?

**Review by ckime-amd — APPROVED** [review]
Minor question about GB vs. GiB, but lgtm otherwise.

File: docs/how-to/system-optimization/mi300a.rst (hunk 1 above)
Comment (amd-jnovotny) [reply docs/how-to/system-optimization/mi300a.rst:128]: Sure. I can make this change.

File: docs/how-to/system-optimization/mi300a.rst [hunk 2]
```diff
@@ -122,6 +122,51 @@ This section describes performance-based settings.
 
        transparent_hugepage=always
 
+* **Increase the amount of allocatable memory**
+
+  By default, when using a device allocator via HIP, it is only possible to allocate 96GiB out of 
+  a possible 128GiB of memory on the MI300A. This limitation does not affect host allocations.
+  To increase the amount of available system memory, load the ``amdttm`` module with new values for
```
Comment (lpaoletti) [inline docs/how-to/system-optimization/mi300a.rst:129]:
```suggestion
  To increase the available system memory, load the ``amdttm`` module with new values for
```

File: docs/how-to/system-optimization/mi300a.rst [hunk 3]
```diff
@@ -122,6 +122,51 @@ This section describes performance-based settings.
...
+
+  .. code-block:: shell
+
+     cat /sys/module/ttm/parameters/pages_limit 
+
+  .. note::
+
+     The system settings for ``pages_limit`` and ``page_pool_size`` are calculated by multiplying the
+     per-APU limit of 4KiB pages, which is ``33554432``, by the number of APUs on the node. For a system with
+     two APUs, the limit is ``33554432 x 2`` or ``67108864``.
```
Comment (lpaoletti) [inline docs/how-to/system-optimization/mi300a.rst:167]:
```suggestion
     per-APU limit of 4KiB pages, which is ``33554432``, by the number of APUs on the node. The limit for a system with
     two APUs is ``33554432 x 2`` or ``67108864``.
```

**Review by lpaoletti — APPROVED** [review]

**Pull request merged**